src/plugins/workspace/public/plugin.ts

```diff
@@ -11,59 +12,59 @@ import {
   AppMountParameters,
   AppNavLinkStatus,
 } from '../../../core/public';
-import { WORKSPACE_APP_ID, WORKSPACE_ID_IN_SESSION_STORAGE, PATHS } from '../common/constants';
+import { WORKSPACE_APP_ID, PATHS } from '../common/constants';
 import { WORKSPACE_ID_QUERYSTRING_NAME } from '../../../core/public';
 import { mountDropdownList } from './mount';
 
 export class WorkspacesPlugin implements Plugin<{}, {}> {
   private core?: CoreSetup;
-  private addWorkspaceListener() {
-    this.core?.workspaces.client.currentWorkspaceId$.subscribe((newWorkspaceId) => {
-      try {
-        sessionStorage.setItem(WORKSPACE_ID_IN_SESSION_STORAGE, newWorkspaceId);
-      } catch (e) {
-        /**
-         * in incognize mode, this method may throw an error
-         * */
-      }
-    });
-  }
   private getWorkpsaceIdFromQueryString(): string | null {
     const searchParams = new URLSearchParams(window.location.search);
     return searchParams.get(WORKSPACE_ID_QUERYSTRING_NAME);
   }
-  private getWorkpsaceIdFromSessionStorage(): string {
-    try {
-      return sessionStorage.getItem(WORKSPACE_ID_IN_SESSION_STORAGE) || '';
-    } catch (e) {
-      /**
-       * in incognize mode, this method may throw an error
-       * */
-      return '';
+  private async getWorkpsaceId(): Promise<string> {
```

Comment:
I may suggest to change `getWorkpsaceId` to return `string` instead of promise. It's not a async operation, we should comply the fact. This function will be used in many places, it forces us to to refactor a regular function to async which I think is unnecessary.

Reply:
This function is a internal method and only be called inside this file.

Reply:
To be more specific, I mean original place: `workspaces.client.getCurrentWorkspaceId`